experiment.py

```diff
@@ -0,0 +1,207 @@
+    
+    def pathology_to_eng(self, pathology: str):
+        return PatientProfile.release_conditions[pathology]["cond-name-eng"]
+
+    def calc_acc(self, dx_regex: str = r"the most likely diagnosis is (.*)\.", count: int = int(1e8), verbose: bool = False) -> float:
+        # Load ground truths
+        labels = [self.pathology_to_eng(pat.PATHOLOGY) for pat in self._samples.itertuples()][:count]
+        count = min(count, len(labels))
+        # Load logged results
+        filenames = list()
+        exist_files = set(os.listdir(self._log_path))
+        for i in range(1, count + 1):
+            filename = f"doctor-{i}.json"
+            if filename in exist_files:
+                filenames.append(filename)
+            else:
+                raise ValueError(f"File {filename} not found.")
+        assert len(labels) == len(filenames)
+        # Extract predictions
+        preds = list()
+        for filename, label in zip(filenames, labels):
+            d = json.loads((self._log_path / filename).read_bytes())
+            dx = ''
+            # logging.info(filename)
+            for doctor_utter in d["doctor_utters"]:
+                dxs = re.findall(pattern=dx_regex, string=doctor_utter)
+                if len(dxs) == 1:
+                    if not dx:
+                        dx = dxs[0]
+                    else:
+                        if (dx != dxs[0]):
+                            # dx = dxs[0]
+                            logging.warning(f"!!! Diagnosis {dx} not equal to {dxs[0]} in file {filename} !!!")
+                elif len(dxs) > 1:
+                    raise ValueError("There shouldn't be more than 1 match of 'dx_regex'. Please check.")
+            if not dx:
+                raise ValueError("There should be at least 1 diagnosis in each dialouge.")
+            preds.append(dx)
+        assert len(preds) == len(labels)
+        # Comparison
+        ncorrects = 0
+        for pred, label in zip(preds, labels):
+            if verbose:
+                    logging.info(f"Ground Truth: {Fore.RED + label + Style.RESET_ALL} / Prediction: {Fore.BLUE + pred + Style.RESET_ALL}{f' {Fore.GREEN}✔{Style.RESET_ALL}' if (pred == label) else ''}")
+            if pred == label:
+                ncorrects += 1
+        return ncorrects / len(preds)
+
+    def run(self, api_interval: int) -> None:
+        """
+            Run the experiment. The interval between API calls is set to [api_interval] seconds.
+        """
+        # TODO: save experiment config (yml)
+        # Cost estimation
+        cost = self.estimate_cost()
+        flag = input(f"Estimated cost: {cost} USD (~{cost * 30} TWD). Continue? (Y)")
+        if flag != 'Y':
+            logging.info("Experiment aborted.")
+            return
+        
+        for i, pat in enumerate(self._samples.itertuples()):
+            if (i < 63): # NOTE: conduct the subset
```

Comment:
What dose 63 means?

Reply:
Haha it is just an API "breakpoint" in an experiment. I forgot to delete.